tests/pdctl/config/config_test.go

```diff
@@ -126,39 +140,34 @@ func (s *configTestSuite) TestConfig(c *C) {
-	c.Assert(err, IsNil)
-	args1 = []string{"-u", pdAddr, "config", "set", "hot-region-schedule-limit", "64"}
-	_, _, err = pdctl.ExecuteCommandC(cmd, args1...)
-	c.Assert(err, IsNil)
-	args1 = []string{"-u", pdAddr, "config", "set", "hot-region-cache-hits-threshold", "5"}
-	_, _, err = pdctl.ExecuteCommandC(cmd, args1...)
-	c.Assert(err, IsNil)
-	args2 = []string{"-u", pdAddr, "config", "show"}
-	_, output, err = pdctl.ExecuteCommandC(cmd, args2...)
-	c.Assert(err, IsNil)
-	cfg = config.Config{}
-	c.Assert(json.Unmarshal(output, &cfg), IsNil)
-	scheduleCfg = cfg.Schedule
-	c.Assert(scheduleCfg.LeaderScheduleLimit, Equals, svr.GetScheduleConfig().LeaderScheduleLimit)
-	c.Assert(scheduleCfg.HotRegionScheduleLimit, Equals, svr.GetScheduleConfig().HotRegionScheduleLimit)
-	c.Assert(scheduleCfg.HotRegionCacheHitsThreshold, Equals, svr.GetScheduleConfig().HotRegionCacheHitsThreshold)
-	c.Assert(scheduleCfg.HotRegionCacheHitsThreshold, Equals, uint64(5))
-	c.Assert(scheduleCfg.HotRegionScheduleLimit, Equals, uint64(64))
-	c.Assert(scheduleCfg.LeaderScheduleLimit, Equals, uint64(64))
-	args1 = []string{"-u", pdAddr, "config", "set", "enable-remove-down-replica", "false"}
-	_, _, err = pdctl.ExecuteCommandC(cmd, args1...)
-	c.Assert(err, IsNil)
-	args2 = []string{"-u", pdAddr, "config", "show"}
-	_, output, err = pdctl.ExecuteCommandC(cmd, args2...)
-	c.Assert(err, IsNil)
-	cfg = config.Config{}
-	c.Assert(json.Unmarshal(output, &cfg), IsNil)
-	scheduleCfg = cfg.Schedule
-	c.Assert(scheduleCfg.EnableRemoveDownReplica, Equals, svr.GetScheduleConfig().EnableRemoveDownReplica)
-	c.Assert(scheduleCfg.EnableRemoveDownReplica, IsFalse)
+	// test config read and write
+	testItems := []testItem{
+		{"leader-schedule-limit", uint64(64), func(scheduleConfig *config.ScheduleConfig) interface{} {
+			return scheduleConfig.LeaderScheduleLimit
+		}}, {"hot-region-schedule-limit", uint64(64), func(scheduleConfig *config.ScheduleConfig) interface{} {
+			return scheduleConfig.HotRegionScheduleLimit
+		}}, {"hot-region-cache-hits-threshold", uint64(5), func(scheduleConfig *config.ScheduleConfig) interface{} {
+			return scheduleConfig.HotRegionCacheHitsThreshold
+		}}, {"enable-remove-down-replica", false, func(scheduleConfig *config.ScheduleConfig) interface{} {
+			return scheduleConfig.EnableRemoveDownReplica
+		}},
+	}
+	for _, item := range testItems {
+		// write
+		args1 = []string{"-u", pdAddr, "config", "set", item.name, reflect.TypeOf(item.value).String()}
+		_, _, err = pdctl.ExecuteCommandC(cmd, args1...)
+		c.Assert(err, IsNil)
+		//read
+		args2 = []string{"-u", pdAddr, "config", "show"}
+		_, output, err = pdctl.ExecuteCommandC(cmd, args2...)
+		c.Assert(err, IsNil)
+		cfg = config.Config{}
+		c.Assert(json.Unmarshal(output, &cfg), IsNil)
+		//judge
+		item.judge(c, &cfg.Schedule, svr.GetScheduleConfig())
+	}
+
+	// test error or deprecated config name
 	args1 = []string{"-u", pdAddr, "config", "set", "foo-bar", "1"}
```

Comment:
Is there possible to make them table-driven?

Reply:
no, it is difficult...